src/chains/base.ts

```diff
@@ -1,12 +1,45 @@
 import { Chain, ChainContract } from 'viem'
 import { base as viemBase } from 'viem/chains'
-import { opStackL2ContractAddresses } from '../types/opStackContracts'
+
+export type OpStackChain = Chain & {
+  contracts: OpStackL2Contracts
+  optimismConfig: OptimismConfig
+}
+
+export type OptimismConfig = {
+  l1: {
+    chainId: number
+    contracts: {
+      [key: string]: ChainContract
+    }
+  }
+}
```

Comment:
@roninjin10 chain.contracts is a thing and is imo where all the L2 stuff should live because they are L2 contracts, so not sure why we need a sub dict inside contracts?

Reply:
Ahhh I think it just clicked what you are going for. l1 is itself a chain